src/models/converters/safes.rs

```diff
@@ -1,10 +1,24 @@
-use crate::models::service::safes::{AddressEx, SafeInfoEx};
+use crate::models::service::safes::{AddressEx, ImplementationVersionState, SafeInfoEx};
 use crate::providers::ext::InfoProviderExt;
 use crate::providers::info::{InfoProvider, SafeInfo};
+use semver::Version;
+use std::cmp::Ordering;
 
 // We need to add Sync as trait bound as info_provider moves across threads
 impl SafeInfo {
     pub async fn to_safe_info_ex(&self, info_provider: &(impl InfoProvider + Sync)) -> SafeInfoEx {
+        let min_chain_version = info_provider
+            .chain_info()
+            .await
+            .ok()
+            .as_ref()
+            .map(|chain_info| chain_info.recommended_master_copy_version.to_string());
+        let implementation_version_state = self
+            .version
+            .as_ref()
+            .map_or(ImplementationVersionState::Unknown, |it| {
+                calculate_version_state(it, &min_chain_version)
+            });
```

Comment:
We return `Unknown` for failure on fetching details. Shouldn't happen, but also, we shouldn't fail the rest of the response in the eventual case.